modules/gfac/gfac-impl/src/main/java/org/apache/airavata/gfac/impl/task/SCPDataStageTask.java

```diff
@@ -111,24 +136,162 @@ public TaskStatus execute(TaskContext taskContext) {
                     return status;
+                } else {
+                    return transferFiles(processContext, subTaskModel, taskContext, processState);
                 }
+
             } else {
                 status.setState(TaskState.FAILED);
                 status.setReason("Invalid task invocation, Support " + ProcessState.INPUT_DATA_STAGING.name() + " and " +
                         "" + ProcessState.OUTPUT_DATA_STAGING.name() + " process phases. found " + processState.name());
                 return status;
             }
 
+        } catch (TException e) {
+            String msg = "Couldn't create subTask model thrift model";
+            log.error(msg, e);
+            status.setState(TaskState.FAILED);
+            status.setReason(msg);
+            ErrorModel errorModel = new ErrorModel();
+            errorModel.setActualErrorMessage(e.getMessage());
+            errorModel.setUserFriendlyMessage(msg);
+            taskContext.getTaskModel().setTaskErrors(Collections.singletonList(errorModel));
+            return status;
+        }
+    }
+
+    private TaskStatus extractStringFromFile(DataStagingTaskModel subTaskModel, TaskContext taskContext, String arguments) {
+
+        TaskStatus status = new TaskStatus(TaskState.EXECUTING);
+
+        File tempFile = null;
+        try {
+            if (arguments != null && !arguments.isEmpty()) {
+
+                String[] allArgs = arguments.split(",");
+                tempFile = File.createTempFile("temp-output", ".tmp");
+                URI sourceURI = new URI(subTaskModel.getSource());
+
+                log.info("Downloading file " + sourceURI.getPath() + " to temporary file " + tempFile);
+
+                taskContext.getParentProcessContext().getJobSubmissionRemoteCluster().copyFrom(sourceURI.getPath(), tempFile.getPath());
+
+                StringBuilder result = new StringBuilder();
+                // this is to identify that output is parsed
+                result.append("parsed-out: ");
+
+                // if the file size is grater than 2 MB, skip parsing to avoid possible OOM issues
+                if (tempFile.length() <= MAX_FILE_SIZE_TO_READ) {
+
+                    try (FileReader fr = new FileReader(tempFile)) {
+                        log.info("Searching for lines that contains " + arguments + " in file " + tempFile);
+                        BufferedReader br = new BufferedReader(fr);
+
+                        String line;
+                        while ((line = br.readLine()) != null) {
+
+                            if (log.isTraceEnabled()) {
+                                log.trace("Input file line : {}", line);
+                            }
+
+                            for (String arg : allArgs) {
+                                if (line.startsWith(arg)) {
+                                    log.debug("Found a line with argument {} : {}", arg, line);
+                                    result.append(line).append("\n");
```

Comment:
Are multiple values supported? Or should this break after finding the first one?